### DISCUSSION TITLE
Sync between Google Tasks and Obsidian

### DISCUSSION THREAD

**Discussion by jmriego** [bergercookie/syncall#121] [Ideas]
Is there any plans to implement sync between Google Tasks and Obsidian?

Obsidian supports creating tasks as simple checklist items in a Markdown file, so this projects has most of the code in place to make this happen.
I can try to create a PR if you'd prefer, but I'm having some issues understanding this. As the GKeep to FS separates items in different files in the filesystem but the sync I'm proposing would actually have all tasks inside the same file. Not sure how to split the file, compare lines and all of that.

Thanks in advance!

**Comment by bergercookie** [comment]
I'm not planning on working on a new integration at the moment.
Having said that, contributions are quite welcome!

There are instructions on how to create such a new integration between services [here](https://github.com/bergercookie/syncall/blob/master/implement-a-new-synchronization.md). If you feel you want more help during your development, just continue posting in this issue and people will respond :)

I'll convert this to a discussion to facilitate this.

**Reply by jmriego** [reply]
I have started giving it a try and I'm a bit stuck trying to understand how to go piece by piece and making sure every class is doing what it's supposed to do. This is the code so far: [Obsidian branch](https://github.com/bergercookie/syncall/compare/master...jmriego:syncall:obsidian)

My intention is to have a Markdown class that points to one single file and that file should be sync to a specific Google Tasks list. Each line in the Markdown file would correspond to a task in that list.

Any hints on how to tackle this? Thanks!

**Comment by jmriego** [comment]
hi @bergercookie ! I started having some code that almost does the work and there's one detail that I'm not sure how to tackle.

I'm syncing between GTasks and MarkDown files now but if I update a Google Task, it will try to update that corresponding Markdown task which is not finding.
The reason is that I assign task ids in Markdown as a hashed of the task description. So if the description changes in Markdown, the old id disappears. I don't want to write the id in the Markdown file itself.

How would you solve that? Do you usually have to track ids between different synchronization sides?

**Comment by jmriego** [comment]
I managed to get this branch mostly working. The only thing I'd like to figure out is how to handle changes in id.

- Let's say I have a Google Task: `{"id": 1, "title": "do something"}`
- and a Markdown Task `{"id": 1001, "title": "do something"}`

The way that I handle Markdown tasks ids is by having it as a hash of the title. So if I update the title in Google Tasks, the id of the Markdown will change as well:
- Google Task: `{"id": 1, "title": "do something changed"}`
- Markdown Task `{"id": 9999, "title": "do something changed"}`

Is there any way I can refresh the Serialized bidict mapping of ids? If I don't, the next run of the sync will delete a task and insert a new one in Google Tasks even if that's not really necessary
